src/zkevm_specs/evm/execution/return_revert.py

```diff
@@ -26,14 +28,21 @@ def return_(instruction: Instruction):
 
     rwc_delta = 3
 
-    if instruction.curr.is_create:
+    if instruction.curr.is_create and is_success:
         # A. Returns the specified memory chunk as deployment code.
 
         # TODO: Untested case.  Test it once create Tx is implemented, and once
         # CREATE/CREATE2 are implemented.
+        callee_address = instruction.call_context_lookup(CallContextFieldTag.CalleeAddress)
+        reversion_info = instruction.reversion_info()
+        code_hash, code_hash_prev = instruction.account_write(
+            callee_address, AccountFieldTag.CodeHash
+        )
+        instruction.constrain_equal(code_hash_prev, RLC(EMPTY_HASH))
+        instruction.constrain_equal(code_hash, instruction.curr.aux_data)
```

Comment:
We should try to have a better name than `aux_data` xD. It's really confusing as it gets compared to a bunch of different stuff.

Reply:
Ah, we're using `aux_data` here as a way to pass auxiliary witness data.  In the halo2 implementation we don't have this because the witness assignment is separate and uses the traces processed by the circuit input builder; but here we're doing constraints and witness together.

Also we're using typed python, but maybe we could find a way to add fields dynamically to the Step object, so that the name of the field matches with the description of it's content?  Nevertheless I suggest approaching this in another PR :)

Reply:
>Also we're using typed python, but maybe we could find a way to add fields dynamically to the Step object, so that the name of the field matches with the description of it's content? Nevertheless I suggest approaching this in another PR :)

Exactly! I was referring to this and the suggestion seems nice!!!

Reply:
I've created an issue for this enhancement https://github.com/privacy-scaling-explorations/zkevm-specs/issues/331
We can now resolve this conversation right?